### PR TITLE
[wip] Lots of files with few simple improvements

### PR DESCRIPTION
**Pull request by Alberth289346**

### CODE REVIEW

File: CorsixTH/Lua/humanoid_actions/walk.lua [hunk 1]
```diff
@@ -112,7 +112,7 @@ action_walk_interrupt = permanent"action_walk_interrupt"( function(action, human
 end)
 
 local flag_list_bottom = 2048
-local flag_early_list = 1024
+--  flag_early_list = 1024
```
Comment (TheCycoONE) [inline CorsixTH/Lua/humanoid_actions/walk.lua]: Why leave this here?

File: CorsixTH/Lua/map.lua [hunk 2]
```diff
@@ -180,8 +180,9 @@ function Map:load(level, difficulty, level_name, map_file, level_intro, map_edit
   local result = file(path .. "Lua" .. pathsep .. "base_config.lua")
 
   local base_config = result
-  local errors
+  local _
```
Comment (TheCycoONE) [inline CorsixTH/Lua/map.lua]: Let's not require declaring (or avoiding declaring) ```_```. It is a near universal convention (mentioned in PIL and enforced by several IDEs) in lua to use _ as a dummy variable so the chance of conflicting with anything else is very slim, which makes this a micro-optimization.

File: CorsixTH/Lua/objects/plant.lua [hunk 3]
```diff
@@ -73,9 +73,6 @@ object.orientations = {
 -- For plant: droop down: 1950
 -- back up again: 1952
 
--- The states specify which frame to show
-local states = {"healthy", "drooping1", "drooping2", "dying", "dead"}
```
Comment (TheCycoONE) [inline CorsixTH/Lua/objects/plant.lua]: This is currently the only documentation for ```current_state```, best to replace it with a comment indicating what each of the current_state values mean.

File: CorsixTH/Lua/humanoid_actions/walk.lua (hunk 1 above)
Comment (Alberth289346) [inline CorsixTH/Lua/humanoid_actions/walk.lua]: Documentation purposes, mostly. I don't know another place where those bit numbers and names/explanation are related with each other.
Note that I also don't know if the complicated way of building the screen is still used/useful; using OpenGL blitting opens a lot of other ways of doing things, like the Z-buffer.

File: CorsixTH/Lua/objects/plant.lua (hunk 3 above)
Comment (Alberth289346) [inline CorsixTH/Lua/objects/plant.lua]: Fair point, replaced this by a line of comment

**Comment by Alberth289346** [comment]
Pushed updated branch (plant state list modified as described)

File: CorsixTH/Lua/map.lua (hunk 2 above)
Comment (TheCycoONE) [inline CorsixTH/Lua/map.lua]: I see, luacheck allows shadowing _, but not referencing it as a global. Fine then, we'll follow their lead.

File: CorsixTH/Lua/humanoid_actions/walk.lua (hunk 1 above)
Comment (TheCycoONE) [inline CorsixTH/Lua/humanoid_actions/walk.lua]: It's still used.

Comment (Alberth289346) [inline CorsixTH/Lua/humanoid_actions/walk.lua]: Removed the line, it's also at objects/door.lua, so the link doesn't get lost

**Comment by Alberth289346** [comment]
Removed the "--  flag_early_list = 1024" line entirely from walk.lua, as another file still has it.

**Comment by Alberth289346** [comment]
Added list with plant state numbers

File: CorsixTH/Lua/objects/plant.lua [hunk 4]
```diff
@@ -70,11 +70,13 @@ object.orientations = {
 -- For litter: put broom back 356
 -- take broom out: 1874
 -- swoop: 1878
--- For plant: droop down: 1950
--- back up again: 1952
+-- Frames for plant states are
```
Comment (TheCycoONE) [inline CorsixTH/Lua/objects/plant.lua]: I didn't mean animation numbers.... ok, maybe you should just document self.current_state (which was an index into the state array)

Comment (Alberth289346) [inline CorsixTH/Lua/objects/plant.lua]: No it is not an index, as it starts at 0. It also ends at 5, so there is a missing index somewhere.
I don't want to debug and fix all plant-related stuff now.

**Pull request merged**